core/src/test/scala/integration/kafka/server/KRaftClusterTest.scala

```diff
@@ -1106,6 +1107,34 @@ class KRaftClusterTest {
       cluster.close()
     }
   }
+
+  @Test
+  def testTimedOutHeartbeats(): Unit = {
+    val cluster = new KafkaClusterTestKit.Builder(
+      new TestKitNodes.Builder().
+        setNumBrokerNodes(3).
+        setNumControllerNodes(1).build()).
+      setConfigProp(KafkaConfig.BrokerHeartbeatIntervalMsProp, 10.toString).
+      setConfigProp(KafkaConfig.BrokerSessionTimeoutMsProp, 1000.toString).
+      build()
+    try {
+      cluster.format()
+      cluster.startup()
+      val controller = cluster.controllers().values().iterator().next()
+      controller.controller.waitForReadyBrokers(3).get()
+      TestUtils.retry(60000) {
+        val latch = controller.controller.asInstanceOf[QuorumController].pause()
+        Thread.sleep(1001)
+        latch.countDown()
+        controller.controller.asInstanceOf[QuorumController].pause().countDown()
```

Comment:
What is the purpose of this line?  Seems it is an unnecessary extra pause/resume, and the test passes without it.  Can we remove it?

Reply:
I'll add a better explanation for this